docs/DeveloperGuide.md

```diff
@@ -158,6 +158,58 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
 
 This section describes some noteworthy details on how certain features are implemented.
 
+### Command History
+
+Command history is a feature that aims to improve the user experience for experienced users by allowing them to quickly
+navigate through their history of commands to make minor changes. Many features were inspired from Macbook's Bash shell
+
+**Implementation**
+An array list was used to store the history of commands and an index to indicate which command is the history currently 
+at. 
+- The list is initialised to have an empty string as the initial element
+- The current command index defaults to 0
+
+There are a few methods used to interact with the command history
+1. getCurrentCommand()
+   1. Gets the command at the current command index
+2. undo()
+   1. Decrements the current command index by 1
+   2. If the current command index is already 0, it will play a Boop sound to indicate that there is 
+are no more commands left to undo
+3. redo()
+   1. Increments the current command index by 1
+
+Below shows expected behaviour of the command history from a series of actions. 
+- blue underline - denotes where the command index
+is pointing at.
+- initial - the initial state of the command history
+- exec - when any command is executed
+- undo - decrements the index
+- redo - increments the index
+
+![](/Users/tkh/Documents/Code/tp/docs/images/command-history/command-history-illustration.jpeg)
```

Comment:
For this illustration, did you include it intentionally or accidentally? Need to use a relative path, please.